### DISCUSSION TITLE
Cannot create URL for blob!

### DISCUSSION THREAD

**Discussion by magicvirus** [supabase/supabase#8653] [Questions]
Following Supa base examples I keep getting cannot create URL for blob!

Expo 46.0.6,  React Native 69.4

const { data, error } = await supabase.storage
        .from('avatars')
        .download(path)
 const url = URL.createObjectURL(data)
      setAvatarUrl(url)

Error cannot create URL for blob!

I can get signed URL working. What's the performance difference of SignedURL vs ObjectURL?

**Comment by GaryAustin1** [comment]
I don't use Expo or React Native, but seem to recall blob issues in discussions/etc.  

One link in a quick Google...
Fetching and Using Blob Data in React Native: Demystified
https://www.waldo.com/blog/react-native-fetch-blob
`If you're using Expo, blob support is still a work in progress. But if you're developing for the React Native CLI, you absolutely can.`

In the storage API reference it mentions issues uploading blobs with React Native.  `For React Native, using either Blob, File or FormData does not work as intended. Upload file using ArrayBuffer from base64 file data instead, see example below.`   but has no warning in download.

I can't help more on the blob front...

On performance:

getSignedUrl does not download the file, just goes and checks RLS and gets a URL with token from the database.  So no storage access.  Then when you use the URL it would download to the device.   

The .download operation would check RLS with a database operation and then go to s3 to download to the device. 

In both cases, the file should be downloaded once from s3, so should be similar.  I don't know if there are differences in how react native might treat the two operations internally.
